jmespath-core/src/main/java/io/burt/jmespath/node/JsonLiteralNode.java

```diff
@@ -3,35 +3,33 @@
 import io.burt.jmespath.Adapter;
 
 public class JsonLiteralNode<T> extends Node<T> {
-  private final String raw;
-  private final T tree;
+  private final String rawJson;
+  private final T json;
 
-  public JsonLiteralNode(Adapter<T> runtime, String raw, T tree) {
+  public JsonLiteralNode(Adapter<T> runtime, String rawJson) {
     super(runtime);
-    this.raw = raw;
-    this.tree = tree;
+    this.rawJson = rawJson;
+    this.json = runtime.parseString(rawJson);
   }
 
   @Override
   public T search(T input) {
-    return tree;
+    return json;
   }
 
   @Override
   protected String internalToString() {
-    return String.format("%s", raw);
+    return String.format("%s", rawJson);
   }
 
   @Override
   protected boolean internalEquals(Object o) {
     JsonLiteralNode<?> other = (JsonLiteralNode<?>) o;
-    return tree.equals(other.tree);
+    return json.equals(other.json);
```

Comment:
Should this not be using the JSON string rather than the value? I guess one could argue both ways, but this way a JSON-literal node with whitespace would be equal to a JSON-literal node without whitespace.